openshift.py

```diff
@@ -0,0 +1,750 @@
+                            + user
+                            + ","
+                            + project_name
+                            + ","
+                            + role
+                            + ","
+                            + op
+                            + ")"
+                        }
+                    ),
+                    status=400,
+                    mimetype="application/json",
+                )
+
+            # now add or remove the user
+            r = self.update_rolebindings(project_name, openshift_role, role_binding)
+
+            msg = "unknown message"
+            if r.status_code == 200 or r.status_code == 201:
+                if op == "add":
+                    msg = "Added role to user on project"
+                elif op == "del":
+                    msg = "removed role from user on project"
+                return Response(
+                    response=json.dumps({"msg": msg}),
+                    status=200,
+                    mimetype="application/json",
+                )
+            if op == "add":
+                msg = "unable to add role to user on project"
+            elif op == "del":
+                msg = "unable to remove role from user on project"
+            return Response(
+                response=json.dumps({"msg": msg}),
+                status=400,
+                mimetype="application/json",
+            )
+        return Response(
+            response=json.dumps(
+                {
+                    "msg": "rolebinding already exists ("
+                    + user
+                    + ","
+                    + project_name
+                    + ","
+                    + role
+                    + ")"
+                }
+            ),
+            status=400,
+            mimetype="application/json",
+        )
+
+
+class openshift_3_x(openshift):
+
+    # member functions for projects
+    def project_exists(self, project_name):
+        url = "https://" + self.get_url() + "/oapi/v1/projects/" + project_name
+        r = self.get_request(url, True)
+        if r.status_code == 200 or r.status_code == 201:
+            return True
+        return False
+
+    def create_project(self, project_name, display_name, user_name):
+        # check project_name
+        url = "https://" + self.get_url() + "/oapi/v1/projects"
+        payload = {
+            "kind": "Project",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": project_name,
+                "annotations": {
+                    "openshift.io/display-name": display_name,
+                    "openshift.io/requester": user_name,
+                },
+            },
+        }
+        r = self.post_request(url, payload, True)
+        return r
+
+    def delete_project(self, project_name):
+        # check project_name
+        url = "https://" + self.get_url() + "/oapi/v1/projects/" + project_name
+        r = self.del_request(url, True)
```

Comment:
You're passing `True` as the request payload here. I suspect you were trying to treat `del_request` like `get_request` (and goes to my earlier comment about delete requests probably not needing a payload argument).

Reply:
Yup, I missed that one.  It was from a time when, I, like you, thought the delete request didn't need a payload, and is a more typical delete request that doesn't require a payload.